clients/spark/core/src/main/scala/io/treeverse/clients/StorageUtils.scala

```diff
@@ -80,5 +87,71 @@ object StorageUtils {
       if (keys.isEmpty) return Seq.empty
       keys.map(x => snPrefix.concat(x))
     }
+
+    def createAndValidateS3Client(
+        configuration: ClientConfiguration,
+        credentialsProvider: Option[AWSCredentialsProvider],
+        awsS3ClientBuilder: AmazonS3ClientBuilder,
+        region: String,
+        bucket: String
+    ): AmazonS3 = {
+
+      require(awsS3ClientBuilder != null)
+      require(bucket.nonEmpty)
+
+      var client =
+        initializeS3Client(configuration, credentialsProvider, awsS3ClientBuilder, region)
+
+      if (!validateClientAndBucketRegionsMatch(client, bucket)) {
+        val bucketRegion = getAWSS3Region(client, bucket)
```

Comment:
Continuing previous comment thread about logging here: I would like this line to contain a log explaining that there was a mismatch and that we got the region from S3.  It should probably follow the general strategy for branch logs: what we did and why (input parameters) and what was the result.  So e.g.
```scala
logger.info(s"Bucket ${bucket} not in region ${region}, discovered it in region ${bucketRegion}")
```